src/cmap/wire_protocol/kill_cursors.ts

```diff
@@ -3,52 +3,28 @@ import { maxWireVersion, collectionNamespace, Callback } from '../../utils';
 import { command, CommandOptions } from './command';
 import { MongoError, MongoNetworkError } from '../../error';
 import type { Server } from '../../sdam/server';
-import type { InternalCursorState } from '../../cursor/core_cursor';
-import type { ClientSession } from '../../sessions';
-
-interface KillCursorOptions {
-  session?: ClientSession;
-  immediateRelease: boolean;
-  noResponse: boolean;
-}
+import type { Long } from '../../bson';
 
 export function killCursors(
   server: Server,
   ns: string,
-  cursorState: InternalCursorState,
+  cursorIds: Long[],
+  options: CommandOptions,
   callback: Callback
 ): void {
   callback = typeof callback === 'function' ? callback : () => undefined;
-
-  if (!cursorState.cursorId) {
-    callback(new MongoError('Invalid internal cursor state, no known cursor id'));
-    return;
+  if (!cursorIds || !Array.isArray(cursorIds)) {
+    throw new TypeError('Invalid list of cursor ids provided: ' + cursorIds);
   }
 
-  const cursorIds = [cursorState.cursorId];
-
   if (maxWireVersion(server) < 4) {
     const pool = server.s.pool;
     const killCursor = new KillCursor(ns, cursorIds);
-    const options: KillCursorOptions = {
-      immediateRelease: true,
-      noResponse: true
-    };
 
-    if (typeof cursorState.session === 'object') {
-      options.session = cursorState.session;
```

Comment:
Are we sure we want to stop passing the session here?

Reply:
Hey, sorry it looks like we lost your comments somewhere (maybe the force push). So what's going on here is that the wire protocol `killCursors` command used to take in a `cursorState` which might have had a session, and now it takes an `options` which may have a session. The `options` are passed all the way down to the `command` method which will use the session there, so there's no need to copy it here any longer